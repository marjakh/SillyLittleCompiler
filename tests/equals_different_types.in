// Array and array

let a = new Array(1);
a[0] = 8;
let b = new Array(1);
b[0] = 8;

if (a == b) {
  print(0);
} else {
  print(1);
}

if (a == a) {
  print(2);
} else {
  print(3);
}

// Array and int

let c = 8;

if (a == c) {
  print(4);
} else {
  print(5);
}

if (c == a) {
  print(6);
} else {
  print(7);
}

// Array and function

function foo() {
}

if (a == foo) {
  print(8);
} else {
  print(9);
}

if (foo == a) {
  print(10);
} else {
  print(11);
}

// Int and function
if (c == foo) {
  print(12);
} else {
  print(13);
}

if (foo == c) {
  print(14);
} else {
  print(15);
}

// Function and function

function bar() {
}

if (foo == bar) {
  print(16);
} else {
  print(17);
}

if (bar == foo) {
  print(18);
} else {
  print(19);
}

// This doesn't work as expected. It's complicated.
// if (foo == foo) {
//   print(20);
// } else {
//   print(21);
// }

