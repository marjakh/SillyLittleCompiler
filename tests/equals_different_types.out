1
2
5
7
9
11
13
15
17
19
